pkg/kube/ready.go

```diff
@@ -222,16 +225,17 @@ func (c *ReadyChecker) isPodReady(pod *corev1.Pod) bool {
 	return false
 }
 
-func (c *ReadyChecker) jobReady(job *batchv1.Job) bool {
+func (c *ReadyChecker) jobReady(job *batchv1.Job) (bool, error) {
 	if job.Status.Failed > *job.Spec.BackoffLimit {
 		c.log("Job is failed: %s/%s", job.GetNamespace(), job.GetName())
-		return false
+		// If a job is failed, it can't recover, so throw an error
+		return false, fmt.Errorf("job is failed: %s/%s", job.GetNamespace(), job.GetName())
```

Comment:
Please make the message consistent as the `c.log("Job is failed: %s/%s", job.GetNamespace(), job.GetName())`
```suggestion
		return false, fmt.Errorf("Job is failed: %s/%s", job.GetNamespace(), job.GetName())
```